build/Makefile

```diff
@@ -495,23 +487,38 @@ push-allocator-image: $(ensure-build-image)
 build-allocator-image: $(ensure-build-image) build-allocator-binary build-licenses build-required-src-dist
 	docker build $(agones_path)/cmd/allocator/ --tag=$(allocator_tag) $(DOCKER_BUILD_ARGS)
 
+# build list of images to include in the manifest for the sidecar image.
+push_manifest = $(sidecar_linux_amd64_tag)
 ifeq ($(WITH_WINDOWS), 1)
+push_manifest += $(foreach windows_version, $(WINDOWS_VERSIONS), $(sidecar_tag)-windows_amd64-$(windows_version))
+endif
+ifeq ($(WITH_ARM64), 1)
+push_manifest += $(sidecar_linux_arm64_tag)
+endif
+
 # push the gameservers sidecar image
-push-agones-sdk-image: push-agones-sdk-linux-image
+push-agones-sdk-image: push-agones-sdk-linux-image-amd64
+ifeq ($(WITH_WINDOWS), 1)
+push-agones-sdk-image: push-agones-sdk-windows
+endif
+ifeq ($(WITH_ARM64), 1)
+push-agones-sdk-image: push-agones-sdk-linux-image-arm64
+endif
 	# Ensure that the sidecar manifest is removed before creating a new one.
 	-DOCKER_CLI_EXPERIMENTAL=enabled docker manifest rm $(sidecar_tag)
+	# Agones should start using the -linux_amd64 suffix for Linux images. This allows multi-arch support in the future.
+	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest create $(sidecar_tag) $(push_manifest)
+	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest push $(sidecar_tag)
+
+push-agones-sdk-windows:
 	# Using docker buildx with foreign OSes is a bit awkward. Since the layers but not the tag
 	# was cached we have to rebuild the image (which should be fast since it's cached).
 	$(MAKE) WINDOWS_DOCKER_PUSH_ARGS=--push build-agones-sdk-image-windows
-	# Agones should start using the -linux_amd64 suffix for Linux images. This allows multi-arch support in the future.
-	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest create $(sidecar_tag) $(sidecar_linux_amd64_tag) $(foreach windows_version, $(WINDOWS_VERSIONS), $(sidecar_tag)-windows_amd64-$(windows_version))
-	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest push $(sidecar_tag)
-else
-push-agones-sdk-image: push-agones-sdk-linux-image
-endif
 
-# push the gameservers sidecar image
-push-agones-sdk-linux-image: $(ensure-build-image)
+push-agones-sdk-linux-image-arm64: $(ensure-build-image)
+	$(MAKE) DOCKER_BUILD_ARGS=--push build-agones-sdk-image-arm64
```

Comment:
Out of curiosity, why is this not just `docker push $(sidecar_linux_arm64_tag)` to mimic what we do for amd64 in the following target?

Reply:
https://github.com/Ludea/agones/pull/1#discussion_r856562539